relay-sampling/src/lib.rs

```diff
@@ -472,6 +492,15 @@ impl SamplingConfig {
     }
 }
 
+/// The User related information in the trace context
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct TraceUserContext {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub segment: Option<String>,
```

Comment:
What if you make this non-optional? You still check for empty strings, but can avoid one more indirection. We should also test for the case where an empty string is sent through `id` or `segment` and ensure it is ignored.

```suggestion
    #[serde(default, skip_serializing_if = "String::is_empty")]
    pub segment: String,
```

Reply:
ok, made segment and id non optional (and check for empty string while matching)